libsolidity/ast/Types.h

```diff
@@ -1329,7 +1329,10 @@ class MappingType: public Type
 	bool operator==(Type const& _other) const override;
 	std::string toString(bool _short) const override;
 	std::string canonicalName() const override;
-	bool canLiveOutsideStorage() const override { return false; }
+	BoolResult canLiveOutsideStorage() const override
+	{
+		return BoolResult::err("Mappings cannot live outside storage.");
```

Comment:
Here is a use of boolresult.

Reply:
The error message is never used. We could go back to bool.

Reply:
Indeed, the error message from `BoolResult` is never used, agree with moving back to `bool`